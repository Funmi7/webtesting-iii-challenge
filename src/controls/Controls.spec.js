// Test away!
import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Controls from './Controls';


afterEach(rtl.cleanup);

let wrapper, toggleLockButton, toggleCloseButton;
const mock = jest.fn();

beforeEach(() => {
    wrapper = rtl.render(<Controls />)
    
})

describe('Control component', () => {

    test('Matches the snapshot!', () => {
        expect(wrapper.container).toMatchSnapshot();
    })  
    
    it('Button toggle the closed state', () => {
    const wrapper =  rtl.render(<Controls locked={false} closed={false} toggleClosed={mock} />)
      const toggleCloseButton = wrapper.queryByTestId('toggleCloseButton'); 
    //    expect(wrapper.queryByText(/open/i)).toBeInTheDocument();
       rtl.fireEvent.click(toggleCloseButton)
       expect(mock).toBeCalled()
       expect(wrapper.queryByText(/open/i)).not.toBeInTheDocument();
       expect(wrapper.queryByText(/closed/i)).toBeInTheDocument();
    })
})

