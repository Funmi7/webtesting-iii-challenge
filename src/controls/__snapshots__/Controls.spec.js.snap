// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control component Matches the snapshot! 1`] = `
<div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="toggleCloseButton"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="toggleLockButton"
    >
      Close Gate
    </button>
  </div>
</div>
`;
